tool/rig/main.go

```diff
@@ -46,6 +47,13 @@ func run() error {
 		cupsertChangeset = Ref(cupsert.Flag("changeset", "name of the changeset").Short('c').Envar(changesetEnvVar).Required())
 		cupsertFile      = cupsert.Flag("file", "file with new resource spec").Short('f').Required().String()
 
+		cupsertConfigMap          = app.Command("configmap", "Upsert configmap in the context of a changeset")
+		cupsertConfigMapChangeset = Ref(cupsertConfigMap.Flag("changeset", "name of the changeset").Short('c').Envar(changesetEnvVar).Required())
+		cupsertConfigMapName      = cupsertConfigMap.Arg("name", "ConfigMap name").Required().String()
+		cupsertConfigMapNamespace = cupsertConfigMap.Flag("resource-namespace", "ConfigMap namespace").Default(rigging.DefaultNamespace).String()
+		cupsertConfigMapFiles     = cupsertConfigMap.Flag("from-file", "files or directories with contents").Strings()
```

Comment:
Why mix file sources with directory sources? - these are totally incompatible (for instance, based on the error from `handleConfigMapFromFileSources` when it is provided a directory as part of the file source list).

If you have `--from-file` and `--from-dir` each with its own format (e.g. `--from-dir` just taking a directory path) and make them options that you can specify multiple times you can mix and match any kind of configuration.